csharp_dotnetcore/13.Basic-Bot-Template/BasicBotAccessors.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+
+namespace Microsoft.BotBuilderSamples
+{
+    /// <summary>
+    /// This class is created as a Singleton and passed into the IBot-derived constructor.
+    ///  - See <see cref="BasicBot"/> constructor for how that is injected.
+    ///  - See the Startup.cs file for more details on creating the Singleton that gets
+    ///    injected into the constructor.
+    /// </summary>
+    public class BasicBotAccessors
```

Comment:
>BasicBotAccessors [](start = 17, length = 17)

I really really really don't like this as a pattern.  It feels like an anti pattern to have a global singleton with all of your accesors on it. 

What is the case for these things not being defined as close to the code that consumes it as possible?  

* DialogState should be defined in the bot where the dialogSet is defined
* GreetingState should be definedin the dialog which does greeting 
etc.

Reply:
I'm not sure I like this pattern that much either. But at least we are consistent so if we want to change it we should be able to do that mechanically across all the samples.

Reply:
Either way.  
We still have to inject the underlying ConversationState/UserState  - if we want to remove accessors, it's not a big runtime hit to recreate accessors on every turn.